File: org.eclipse.winery.provenance/src/main/java/org/eclipse/winery/provenance/config/Configuration.java
/*******************************************************************************
 * Copyright (c) 2018 Contributors to the Eclipse Foundation
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
 * which is available at https://www.apache.org/licenses/LICENSE-2.0.
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *******************************************************************************/
package org.eclipse.winery.provenance.config;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public class Configuration {

    private static Configuration instance;
    private static final String PROPERTIES_FILE = "config.properties";
    public Properties properties = null;

    private Configuration() {

    }

    private void readProperties() {
        properties = new Properties();
        final InputStream inputStream = getClass().getClassLoader().getResourceAsStream(PROPERTIES_FILE);
        if (inputStream != null) {
            try {
                properties.load(inputStream);
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }

    public static Configuration getInstance() {
        if (instance == null) {
            instance = new Configuration();
            instance.readProperties();
        }

        return instance;
    }
}

File: org.eclipse.winery.frontends/app/tosca-management/src/app/configuration.ts
/*******************************************************************************
 * Copyright (c) 2017-2020 Contributors to the Eclipse Foundation
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
 * which is available at https://www.apache.org/licenses/LICENSE-2.0.
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *******************************************************************************/
const wineryContext = '/winery';
// when running in development mode on port 4200, use default port 8080
// otherwise, assume that backend runs on the same port#
export const hostURL = location.protocol + '//' + location.hostname + ':' + (location.port === '4200' ? '8080' : location.port);
export const editorURL = location.protocol + '//' + location.hostname + ':' + (location.port === '4200' ? '4201' : location.port + '/winery-topologymodeler');
export const backendBaseURL = hostURL + wineryContext;
export const webSocketURL = 'ws://' + location.hostname + ':' + (location.port === '4200' ? '8080' : location.port) + wineryContext;

const positionOfFirstColon = backendBaseURL.indexOf(':');
const positionOfSecondColon = backendBaseURL.indexOf(':', positionOfFirstColon + 1);
  // example: http://localhost:4242
  // positionOfFirstColon:4, positionOfSecondColon:16
  // same as the following?
  // export const modelerURL = location.protocol + '//' + location.hostname + ':' + '4242';
export const modelerURL = backendBaseURL.substring(0, positionOfSecondColon + 1) + '4242';
